packages/astro/src/core/build/internal.ts

```diff
@@ -80,17 +79,15 @@ export function trackPageData(
 export function trackClientOnlyPageDatas(
 	internals: BuildInternals,
 	pageData: PageBuildData,
-	clientOnlys: string[],
-	astroConfig: AstroConfig
+	clientOnlys: string[]
 ) {
 	for (const clientOnlyComponent of clientOnlys) {
-		const coPath = viteID(new URL('.' + clientOnlyComponent, astroConfig.root));
 		let pageDataSet: Set<PageBuildData>;
-		if (internals.pagesByClientOnly.has(coPath)) {
-			pageDataSet = internals.pagesByClientOnly.get(coPath)!;
+		if (internals.pagesByClientOnly.has(clientOnlyComponent)) {
```

Comment:
`transform()` is now using the module ID directly rather than the project root relative path, no need to convert to here with `viteID`